src/storm/mesos/MesosCommon.java

```diff
@@ -39,6 +40,22 @@
   public static final String SUPERVISOR_ID = "supervisorid";
   public static final String ASSIGNMENT_ID = "assignmentid";
 
+  
+  public static String ipAddressFromAssignmentId(String assignmentId) {
+        int last = assignmentId.lastIndexOf("-");
+        String ip = assignmentId.substring(last + 1);
+        LOG.info("IPAddress: " + ip);
```

Comment:
Need more time to read and test these changes (I think this might in fact be incorrect for my system).  Can you tell me an example value of assignmentId that you are trying to parse the IP out of?   (Also you would *definitely* want more context in this log message, and I suspect it should be debug too.)